src/test/java/java_dz_4/DoctorControllerTest.java

```diff
@@ -0,0 +1,148 @@
+package java_dz_4;
+
+
+import org.assertj.core.api.Assertions;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.util.ResourceUtils;
+
+import java.io.File;
+import java.nio.file.Files;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
+
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+@AutoConfigureMockMvc
+public class DoctorControllerTest {
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    DoctorRepo doctorRepo;
+
+    @After
+    public void cleanUp() {
+        doctorRepo.cleanAll();
+
+    }
+
+
+    @Test
+    public void shouldFindAllDoctors() throws Exception {
+        doctorRepo.createDoctor(new Doctor(null, "Kirill", "sp1"));
+        doctorRepo.createDoctor(new Doctor(null, "Vasya", "sp2"));
+        doctorRepo.createDoctor(new Doctor(null, "Nikita", "sp3"));
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/doctors"))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().json(fromResource("all-doctors.json"), false))
+                .andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(3)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].name", Matchers.is("Kirill")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].specialization", Matchers.is("sp1")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[1].name", Matchers.is("Vasya")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[1].specialization", Matchers.is("sp2")));
```

Comment:
If you add a static import for `MockMvcResultMatchers` and `MockMvcRequestBuilders` code would look much prettier